### DISCUSSION TITLE
Should we validate dependency names, like KeyVault names, secret-names, etc...

### DISCUSSION THREAD

**Discussion by fgheysels** [arcus-azure/arcus.observability#260] [Ideas]
I've seen that the `LogAzureKeyVaultDependency` method verifies if the `secretName` and `keyvaultUri` parameters have a value that is valid as secret-name or KeyVault Uri.   If they're not valid, an exception is thrown.

I wonder if we should do that, for the following reasons:

- Additional responsability lies now wih the Arcus library to make sure that we keep this validation-logic up-to-date.   Suppose the naming rules change in the future, we must be quick to update our library
- In case of the observability lib, we only use this information for adding that information in our traces.   Should we throw exceptions when the information that is passed, represent names that do not represent a valid secretname or keyvault uri ?  I don't think it is a problem if that happens ?   Moreover, should we allow that an application can crash because of this ?

I think we should not validate this kind of stuff.

**Comment by stijnmoreels** [comment]
I'm in favor of removing these validation checks. 🙋‍♂️

**Reply by fgheysels** [reply]
Me too!

@pim-simons  @ALaks96  @gverstraete  Do you guys have an opinion on this ?

**Reply by pim-simons** [reply]
I agree, also in favor to remove the validation checks.

**Reply by gverstraete** [reply]
You could also make this validation extensable...

**Reply by stijnmoreels** [reply]
Yes, that's also an option. But the limited injections point here would make that rather hard, I think. Maybe Serilog has some extra post-validation for this, but not sure if we should go that far.

**Comment by fgheysels** [comment]
We can conclude that we agree that we should remove these validations.

Therefore, an issue has been created for this: https://github.com/arcus-azure/arcus.observability/issues/287
